packages/core/utils/src/auth/token.ts

```diff
@@ -1,21 +1,27 @@
-import jwt from "jsonwebtoken"
+import jwt, { type Secret, type SignOptions } from "jsonwebtoken"
 import { MedusaError } from "../common"
 
 export const generateJwtToken = (
   tokenPayload: Record<string, unknown>,
   jwtConfig: {
-    secret: string | undefined
-    expiresIn: string | undefined
+    secret?: Secret
+    expiresIn?: number | string
+    jwtOptions?: SignOptions
   }
 ) => {
-  if (!jwtConfig.secret || !jwtConfig.expiresIn) {
+  if (
+    !jwtConfig.secret ||
+    (!jwtConfig.expiresIn && !jwtConfig.jwtOptions?.expiresIn)
+  ) {
     throw new MedusaError(
       MedusaError.Types.INVALID_ARGUMENT,
       "JWT secret and expiresIn must be provided when generating a token"
     )
   }
 
```

Comment:
Just for sake of awareness, I would at least `warn` the user if a public key is present on the `jwtConfig` but the options doesnt not contain the `algorithm`. I'm affraid people will just set the public key, but the default algorithm will be symetric.